src/main/java/io/github/notstirred/chunkyeditor/state/vanilla/VanillaStateTracker.java

```diff
@@ -24,6 +24,12 @@ public class VanillaStateTracker {
 
     public VanillaStateTracker(Path regionDirectory) {
         this.regionDirectory = regionDirectory;
+
+        // Hack needed on Windows to get scratch files to delete themselves properly
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            removeAllStates();
+            System.gc();
+        }));
     }
 
```

Comment:
Not really sure about this, as shutdown hooks don't run on a non-standard shutdown of the jvm, and this lambda will hold onto all `VanillaStateTracker` objects until shutdown.

Thoughts on adding a `State#release()` which instructs a state to free any resources?
Then there could just be a single `Files.createTempDirectory` for all of the temp files, and only a single `deleteOnExit` for the temp dir